From e1986ae3e977550d0f86e4244ae23dac48be93c0 Mon Sep 17 00:00:00 2001
From: Kody A Kantor <kody.kantor@gmail.com>
Date: Thu, 26 Oct 2017 15:22:25 +0000
Subject: [PATCH] joyent/node-fast#11 add RPC method name to metrics Reviewed
 by: David Pacheco <dap@joyent.com> Reviewed by: Cody Peter Mello
 <cody.mello@joyent.com> Approved by: David Pacheco <dap@joyent.com>

---
 CHANGES.md         | 3 +++
 lib/fast_server.js | 7 +++++--
 package.json       | 4 ++--
 3 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/CHANGES.md b/CHANGES.md
index 74bed16..2bb4b05 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -4,6 +4,9 @@
 
 No changes.
 
+## v2.3.1
+* #11 add RPC method name to metrics
+
 ## v2.3.0
 
 * #9 node-fast could track basic request metrics
diff --git a/lib/fast_server.js b/lib/fast_server.js
index b82c23b..66e0485 100644
--- a/lib/fast_server.js
+++ b/lib/fast_server.js
@@ -848,6 +848,7 @@ FastServer.prototype.requestCleanup = function (request)
 	var self = this;
 	var diff;
 	var latency;
+	var labels;
 
 	mod_assertplus.equal(request.fsr_state, FR_S_COMPLETE);
 	conn = request.fsr_conn;
@@ -865,8 +866,10 @@ FastServer.prototype.requestCleanup = function (request)
 		diff = process.hrtime(request.fsr_hrtstarted);
 		latency = mod_jsprim.hrtimeMillisec(diff);
 
-		this.fs_request_counter.increment();
-		this.fs_latency_histogram.observe(latency);
+		/* Track the requested RPC methoad. */
+		labels = { 'rpcMethod': request.fsr_rpcmethod };
+		this.fs_request_counter.increment(labels);
+		this.fs_latency_histogram.observe(latency, labels);
 	}
 
 	if (request.fsr_handler !== null) {
diff --git a/package.json b/package.json
index e34bb05..1fc6448 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
 	"name": "fast",
 	"description": "streaming JSON RPC over TCP",
-	"version": "2.3.0",
+	"version": "2.3.1",
 	"main": "./lib/fast.js",
 	"repository": {
 		"type": "git",
@@ -23,7 +23,7 @@
 		"verror": "^1.7.0"
 	},
 	"devDependencies": {
-		"artedi": "1.0.0",
+		"artedi": "1.1.1",
 		"forkexec": "^1.0.0",
 		"restify": "5.2.0"
 	},
-- 
2.21.0

